src/openrct2/platform/Windows.cpp

```diff
@@ -222,15 +221,14 @@ bool platform_get_steam_path(utf8* outPath, size_t outSize)
         return false;
     }
 
-    wSteamPath = (wchar_t*)malloc(size);
-    result = RegQueryValueExW(hKey, L"SteamPath", nullptr, &type, (LPBYTE)wSteamPath, &size);
+    auto wSteamPath = std::make_unique<wchar_t[]>(size);
```

Comment:
This `size` refer to the number of bytes to fit `wchar_t` elements. So either this was a bug in the previous code or this should be called with `size/sizeof(wchar_t)`. Please investigate a bit where this `size` comes from and if it is already adjusted to  `wchar_t`.

Reply:
It seems that it stores the size of the data in bytes, so it might be a bug.